src/Nethermind/Nethermind.Trie/PatriciaTree.cs

```diff
@@ -536,21 +533,26 @@ private void ResolveNode(TrieNode node, in TraverseContext traverseContext)
             }
             catch (TrieNodeException e)
             {
-                ThrowMissingTrieNodeException(in traverseContext, e);
+                ThrowMissingTrieNodeException(e, in traverseContext);
             }
         }
 
-        private CappedArray<byte> TraverseNode(TrieNode node, in TraverseContext traverseContext)
+        private ref readonly CappedArray<byte> TraverseNode(TrieNode node, scoped in TraverseContext traverseContext)
         {
             if (_logger.IsTrace) Trace(node, traverseContext);
 
-            return node.NodeType switch
-            {
-                NodeType.Branch => TraverseBranch(node, in traverseContext),
-                NodeType.Extension => TraverseExtension(node, in traverseContext),
-                NodeType.Leaf => TraverseLeaf(node, in traverseContext),
-                NodeType.Unknown => TraverseUnknown(node),
-                _ => ThrowNotSupported(node)
+            switch (node.NodeType)
```

Comment:
you can't have switch expressions with ref?

Reply:
Not that I can get to compile